File: src/quark/designdocs/orderbook.design/views/bid/map.cpp
/*
 * map.cpp
 *
 *  Created on: Jun 4, 2017
 *      Author: ondra
 */

#include <couchcpp/api.h>

void mapdoc(Document doc) {

	if (doc.getID().substr(0,2) == "o."
			&& doc["finished"].getBool() == false
			&& doc["dir"] == "buy"
			&& doc["status"].getString() == "active"
			&& (doc["type"].getString() == "limit"
				|| doc["type"].getString() == "postlimit")) {

		emit(doc["limitPrice"], doc["size"]);
	}

}

File: src/quark/designdocs/orderbook.design/views/all/map.js
function(doc) {
	if (doc._id.substr(0,2) == "o."
		&& !doc.finished
		&& doc.status == "active"
		&& (doc.type == "limit" || doc.type == "postlimit")) {
			emit([doc.dir=="buy"?0:1,doc.limitPrice], doc.size);
		}	
}